sdk/identity/azure-identity/inc/azure/identity/chained_token_credential.hpp

```diff
@@ -55,7 +60,14 @@ namespace Azure { namespace Identity {
         Core::Context const& context) const override;
 
   private:
+    explicit ChainedTokenCredential(
+        Sources sources,
+        std::string const& enclosingCredential,
+        std::vector<std::string> sourceDescriptions);
+
     Sources m_sources;
+    std::vector<std::string> m_sourceDescriptions;
```

Comment:
Why have a source description and not simply use the friendly name of the credential?

Reply:
That is because the other parameter is called "sources", as well as the typedef is called Sources so I'm keeping the same terminology. I'll rename it to `m_sourcesFriendlyNames`

Reply:
#4418

Reply:
To clarify, what I was asking for was to have all information about a Source contained with the type `Source`.